sdk/spring/spring-cloud-azure-feature-management/src/test/java/com/azure/spring/cloud/feature/management/validationstests/ValidationsTest.java

```diff
@@ -0,0 +1,140 @@
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(SpringExtension.class)
+public class ValidationsTest {
+    @Mock
+    private ApplicationContext context;
+    @Mock
+    private FeatureManagementConfigProperties configProperties;
+
+    private final ObjectMapper OBJECT_MAPPER = JsonMapper.builder()
+        .configure(MapperFeature.ACCEPT_CASE_INSENSITIVE_PROPERTIES, true).build();
+    private final String TEST_CAE_FOLDER_PATH = "validations-tests";
+    private final String INPUTS_USER = "user";
+    private final String INPUTS_GROUPS = "groups";
+    private final String SAMPLE_FILE_NAME_FILTER = "sample";
+    private final String TESTS_FILE_NAME_FILTER = "tests";
+
+    @BeforeEach
+    public void setup() {
+        MockitoAnnotations.openMocks(this);
+        when(configProperties.isFailFast()).thenReturn(true);
+        when(context.getBean(Mockito.contains("TimeWindow"))).thenReturn(new TimeWindowFilter());
+    }
+
+    @AfterEach
+    public void cleanup() throws Exception {
+        MockitoAnnotations.openMocks(this).close();
+    }
+
+    private boolean hasException(ValidationTestCase testCase) {
+        final String exceptionStr = testCase.getIsEnabled().getException();
+        return exceptionStr != null && !exceptionStr.isEmpty();
+    }
+
+    private boolean hasInput(ValidationTestCase testCase) {
+        final LinkedHashMap<String, Object> inputsMap = testCase.getInputs();
+        return inputsMap != null && !inputsMap.isEmpty();
+    }
+
+    private File[] getFileList(String folderPath, String fileNameFilter) {
+        final URL folderUrl = Thread.currentThread().getContextClassLoader().getResource(folderPath);
+        final File folderFile = new File(folderUrl.getFile());
+        return folderFile.listFiles(pathname -> pathname.getName().toLowerCase().contains(fileNameFilter));
+    }
+
+    private List<ValidationTestCase> readTestcasesFromFile(File testFile) throws IOException {
+        final String jsonString = Files.readString(testFile.toPath());
+        final CollectionType typeReference =
+            TypeFactory.defaultInstance().constructCollectionType(List.class, ValidationTestCase.class);
+        return OBJECT_MAPPER.readValue(jsonString, typeReference);
+    }
+
+    private LinkedHashMap<String, Object> readConfigurationFromFile(File sampleFile) throws IOException {
+        final String jsonString = Files.readString(sampleFile.toPath());
+        final LinkedHashMap<String, Object> configurations = OBJECT_MAPPER.readValue(jsonString, new TypeReference<>() {
+        });
+        final Object featureManagementSection = configurations.get("feature_management");
+        if (featureManagementSection.getClass().isAssignableFrom(LinkedHashMap.class)) {
+            return (LinkedHashMap<String, Object>) featureManagementSection;
```

Comment:
Something seems to have changed recently in the build pipelines. Currently any warnings are making builds fail. Add a `@SuppressWarnings("unchecked")` to this for now.

Reply:
Updated